poethepoet/plugin.py

```diff
@@ -0,0 +1,77 @@
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+from typing import Iterator
+
+import tomlkit
+
+try:
+    # pylint: disable=import-error
+    from cleo.commands.command import Command
+    from poetry.console import application as app
+    from poetry.plugins.application_plugin import ApplicationPlugin
+except ModuleNotFoundError:
+    # Hacky defaults to
+    # tell whether poetry is installed or not,
+    # please mypy,
+    # and please Poetry's plugin system (simulate the ApplicationPlugin object)
+    app = type("application", (), {"Application": None})  # pylint: disable=C0103
+    ApplicationPlugin = type(
+        "ApplicationPlugin",
+        (),
+        {"PLUGIN_API_VERION": "1.0.0", "type": "application.plugin"},
+    )
+
+from . import config
+
+
+def get_poe_task_names() -> Iterator[str]:
+    return (
+        name
+        for name in tomlkit.loads(
+            Path(config.PoeConfig().find_pyproject_toml()).read_text()
```

Comment:
I haven't tried it but looking through the code it looks like there's a way to get the pyproject.toml content loaded by poetry without loading it again. Something like `app.config.config`

Reply:
Interesting. But I wouldn't use anything undocumented. Made a discussion [here](https://github.com/python-poetry/poetry/discussions/4930)